### ISSUE TITLE
Improve behaviour when photoview is a child of Hero

### ISSUE THREAD

**Issue by asinel** [renancaraujo/photo_view#15]
When photoview is a child of hero, behaviour is very strange: it blinks before loading an image

**Comment by olive009** [comment]
I got this issue too, but it's OK if using normal FadeInImage.memoryNetwork()

**Comment by krwrang** [comment]
Ran into this as well.
Master page is basically StaggeredGridView -> GestureDetector -> Hero -> FadeInImage

GestureDetector opens detail view, that has:
Center -> Hero -> PhotoView -> NetworkImage
Detail view blinks when opening.

**Comment by olive009** [comment]
one possible implementation:
```
import 'dart:async';
import 'package:flutter/material.dart';
import 'package:photo_view/photo_view_image_wrapper.dart';
import 'package:photo_view/photo_view_scale_boundaries.dart';
import 'package:photo_view/photo_view_scale_state.dart';
import 'package:photo_view/photo_view_utils.dart';
import 'package:after_layout/after_layout.dart';

export 'package:photo_view/photo_view_scale_boundary.dart';

class HeroPhotoView extends StatefulWidget {
  final ImageProvider imageProvider;
  final Widget loadingChild;
  final Color backgroundColor;
  final minScale;
  final maxScale;
  final bool gaplessPlayback;
  final Size size;

  HeroPhotoView({
    Key key,
    @required this.imageProvider,
    this.loadingChild,
    this.backgroundColor = const Color.fromRGBO(0, 0, 0, 1.0),
    this.minScale,
    this.maxScale,
    this.gaplessPlayback = false,
    this.size,
  }) : super(key: key);

  @override
  State<StatefulWidget> createState() {
    return new _HeroPhotoViewState();
  }
}

class _HeroPhotoViewState extends State<HeroPhotoView> {
  PhotoViewScaleState _scaleState;
  GlobalKey containerKey = new GlobalKey();
  ImageInfo _imageInfo;

  Future<ImageInfo> _getImage() {
    Completer completer = new Completer<ImageInfo>();
    ImageStream stream = widget.imageProvider.resolve(new ImageConfiguration());
    var listener = (ImageInfo info, bool completed) {
      completer.complete(info);
      setState(() {
        _imageInfo = info;
      });
    };
    stream.addListener(listener);
    completer.future.then((_) {
      stream.removeListener(listener);
    });
    return completer.future;
  }

  void onDoubleTap() {
    setState(() {
      _scaleState = nextScaleState(_scaleState);
    });
  }

  void onStartPanning() {
    setState(() {
      _scaleState = PhotoViewScaleState.zooming;
    });
  }

  @override
  void initState() {
    _getImage();
    super.initState();
    _scaleState = PhotoViewScaleState.contained;
  }

  @override
  Widget build(BuildContext context) {
    if (_imageInfo == null) {
      return new SizedBox();
    }
    return new PhotoViewImageWrapper(
      onDoubleTap: onDoubleTap,
      onStartPanning: onStartPanning,
      imageProvider: widget.imageProvider,
      imageInfo: _imageInfo,
      scaleState: _scaleState,
      backgroundColor: widget.backgroundColor,
      gaplessPlayback: widget.gaplessPlayback,
      size: widget.size ?? MediaQuery.of(context).size,
      scaleBoundaries: new ScaleBoundaries(
        widget.minScale ?? 0.0,
        widget.maxScale ?? 100000000000.0,
        imageInfo: _imageInfo,
        size: widget.size ?? MediaQuery.of(context).size,
      ),
    );
  }

  Widget buildLoading() {
    return widget.loadingChild != null
        ? widget.loadingChild
        : new Center(
            child: new Text(
            "Loading",
            style: new TextStyle(color: Colors.white),
          ));
  }
}

class PhotoViewInline extends StatefulWidget {
  final ImageProvider imageProvider;
  final Widget loadingChild;
  final Color backgroundColor;
  final minScale;
  final maxScale;

  PhotoViewInline({
    Key key,
    @required this.imageProvider,
    this.loadingChild,
    this.backgroundColor = const Color.fromRGBO(0, 0, 0, 1.0),
    this.minScale,
    this.maxScale,
  }) : super(key: key);

  @override
  State<StatefulWidget> createState() => new _PhotoViewInlineState();
}

class _PhotoViewInlineState extends State<PhotoViewInline>
    with AfterLayoutMixin<PhotoViewInline> {
  Size _size;

  @override
  void afterFirstLayout(BuildContext context) {
    setState(() {
      _size = context.size;
    });
  }

  @override
  Widget build(BuildContext context) {
    return new HeroPhotoView(
      imageProvider: widget.imageProvider,
      loadingChild: widget.loadingChild,
      backgroundColor: widget.backgroundColor,
      minScale: widget.minScale,
      maxScale: widget.maxScale,
      size: _size,
    );
  }
}
```

**Comment by renancaraujo** [comment]
Thanks for reporting that, I will look into that in the next few days.
@olive009 Feel free to open a PR with your solution, this way we can collaborate more easily 😊.

**Comment by renancaraujo** [comment]
Adding another Photo view variant class as suggested by @olive009 in #19 would increase the complexity when using the Photo View API (ie: what if photoViewinline with hero?). In #22 I've added another option to `PhotoView` constructor: `heroTag`. 

Now, to achieve the hero effect, **do not put `PhotoView` as a child of a `Hero`**. Instead, inform a `heroTag` as a param.
Example of usage in a transition between screen1 and screen2:
```dart
// screen1
...
Hero(
  tag: "someTag",
  child: Image.asset(
      "assets/large-image.jpg",
      width: 150.0
  ),
)

// screen2
...
child: PhotoView(
    imageProvider: AssetImage("assets/large-image.jpg"),
    heroTag: "someTag",
)
```


Result:
![heroanim](https://user-images.githubusercontent.com/6718144/44302613-fde97e00-a302-11e8-8aa0-5b4297bb918b.gif)



This should be available in 0.0.6
